cpp/src/arrow/flight/types.h

```diff
@@ -716,5 +720,130 @@ class ARROW_FLIGHT_EXPORT SimpleResultStream : public ResultStream {
   size_t position_;
 };
 
+/// \defgroup flight-error Error Handling
+/// Types for handling errors from RPCs.  Flight uses a set of status
+/// codes standardized across Flight implementations, so these types
+/// let applications work directly with those codes instead of having
+/// to translate to and from Arrow Status.
+
+/// \brief Abstract status code for an RPC as per the Flight
+///   specification.
```

Comment:
How come this is different from FlightStatusCode? Do we want to deprecate the latter?

Reply:
FlightStatusCode was a kludge in the original implementation to add to the arrow::StatusCode list without modifying StatusCode. This was refactored out of transport.h and intends to be definitive. (Both gRPC and UCX backends already translate to it, then the synchronous API translates that into arrow::Status + FlightStatusCode). I would rather make a clean break than try to repurpose or extend FlightStatusCode (which also contains statuses that do not actually exist in Flight on the wire.)